logic.lua

```diff
@@ -1,167 +1,406 @@
---local MAX_GROUPS = 8
```

Comment:
@Byron-Miles What do you say if we move assignment strategies to a separate files?

Reply:
That was my original plan... but then I couldn't really think of anything outside of the two existing logic types and it was just easier to lump them in the same file.

Reply:
Let's do that anyway, to decouple strategy from controller. Once again, having a file with 400 lines is too much.